asciidoctorj-core/src/test/groovy/org/asciidoctor/extension/TableCreatorTreeProcessor.groovy

```diff
@@ -10,41 +10,43 @@ import org.asciidoctor.ast.Table
 @CompileStatic
 class TableCreatorTreeProcessor extends Treeprocessor {
 
+    static JRubyProcessor processor = new JRubyProcessor()
```

Comment:
It is problematic to have this static, if we register the same `TableCreatorTreeProcessor` with two different Asciidoctor instances.
Every Asciidoctor instance has its own Ruby instance.
Therefore if we have it static here, the nodes will be created only in one Ruby instance, and somehow transition from one Ruby instance into the other.
This might work most of the time, but not always.
There are a few subtle differences when using objects that come from 2 Rubys, for example Symbols are no longer the same.

Reply:
I would prefer if the `create...()` methods would continue to just work without having to manually create a JRubyProcessor or JSProcessor.
The ExtensionRegistry could inject a node factory at registration time like it does with the config.
That way it would be completely transparent in which environment the processor is running, and maybe it would even work to reuse the same processor in both environments.

Reply:
I completely agree but I didn't know how to work around this issue.

> The ExtensionRegistry could inject a node factory at registration time like it does with the config.

You mean in the `ExtensionGroupImpl` ? I think we can use the proxy to define the processor but I hope that Groovy won't complain.

Reply:
Actually it's unnecessary :wink: 
I was confused by Groovy but as long as we don't use `default` methods, it's working!